src/DGtal/shapes/SurfaceMesh.h

```diff
@@ -0,0 +1,773 @@
+     - @ref TriangulatedSurface represents 2-manifold triangulated
+     meshes, possibly with boundaries, and provides traversal and
+     neighborhood operations (underlying fast index representation);
+     - @ref PolygonalSurface represents 2-manifold polygonal
+     meshes, possibly with boundaries, and provides traversal and
+     neighborhood operations (underlying fast index representation);
+     - @ref SurfaceMesh represents arbitrary sets of faces with
+     possible connexions between faces, edges and vertices, provides
+     traversal and neighborhood operations (underlying fast index
+     representation), and may represent non manifold meshes.
+     - @ref DigitalSurface is a common interface to represent
+     arbitrary digital surfaces, i.e. boundaries of sets of voxels
+     or interfaces between such sets;
+     - @ref IndexedDigitalSurface is a common interface to represent
+     arbitrary digital surfaces, i.e. boundaries of sets of voxels
+     or interfaces between such sets, but with underlying fast index
+     representation.
+
+     See also SurfaceMeshReader and SurfaceMeshWriter for input/output
+     operations for SurfaceMesh.
+
+     @tparam TRealPoint an arbitrary model of 3D RealPoint.
+     @tparam TRealVector an arbitrary model of 3D RealVector.
+  */
+  template < typename TRealPoint, typename TRealVector >
+  struct SurfaceMesh
+  {
+    typedef TRealPoint                              RealPoint;
+    typedef TRealVector                             RealVector;
+    typedef SurfaceMesh< RealPoint, RealVector >    Self;
+    
+    static const Dimension dimension = RealPoint::dimension;
+    BOOST_STATIC_ASSERT( ( dimension == 3 ) );
+    
+    typedef typename RealVector::Component          Scalar;
+    typedef std::vector<Scalar>                     Scalars;
+    /// The type for counting elements.
+    typedef std::size_t                             Size;
+    /// The type used for numbering vertices and faces
+    typedef std::size_t                             Index;
+    typedef Index                                   Face;
+    typedef Index                                   Edge;
+    typedef Index                                   Vertex;
+    typedef std::pair< Edge, Scalar >               WeightedEdge;
+    typedef std::pair< Face, Scalar >               WeightedFace;
+    /// The type that defines a range of vertices
+    typedef std::vector< Vertex >                   Vertices;
+    typedef std::vector< Vertex >                   VertexRange;
+    /// The type that defines a range of faces
+    typedef std::vector< Edge >                     Edges;
+    typedef std::vector< Edge >                     EdgeRange;
+    typedef std::vector< WeightedEdge >             WeightedEdges;
+    typedef std::vector< WeightedEdge >             WeightedEdgeRange;
+    typedef std::vector< Face >                     Faces;
+    typedef std::vector< Face >                     FaceRange;
```

Comment:
Same here

Reply:
The convention "SomeTypes" vs. "SomeTypeRange" may not be necessary and could confuse the reader

Reply:
do you have a preference (for the one to keep) ?

Reply:
According to other classes, maybe "SomeTypes" (e.g. Vertices)

Reply:
Agreed. I also prefer Vertices to VertexRange. The sole problem is when you have a range of Vertices, Verticess is ugly.

Reply:
Do you have a type for a range of Vertices in your class?